.travis.yml

```diff
@@ -0,0 +1,177 @@
+          update: true
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-trusty-6.0
+          packages:
+            - cmake
+            - cmake-data
+            - clang-6.0
+            - libboost-all-dev
+            - build-essential
+      env:
+        - MATRIX_EVAL="CC=/usr/local/clang-5.0.0/bin/clang && CXX=/usr/local/clang-5.0.0/bin/clang++"
+        - BUILD_TYPE="Debug"
+        - BUILD_FLAGS="-fsanitize=address"
+        - CMAKE_FLAGS=""
+        - TEST_TASK="test"
+    - name: "Linux gcc debug fsanitize"
+      os: linux
+      compiler:
+        - gcc
+      sudo: required
+      addons:
+        apt:
+          update: true
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-7
+            - cmake
+            - cmake-data
+            - libboost-all-dev
+            - build-essential
+      env:
+        - MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
+        - BUILD_TYPE="Debug"
+        - BUILD_FLAGS="-fsanitize=address -fuse-ld=gold"
+        - CMAKE_FLAGS=""
+        - TEST_TASK="test"
+    - name: "Linux clang release"
+      os: linux
+      compiler:
+        - clang
+      sudo: required
+      addons:
+        apt:
+          update: true
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-trusty-6.0
+          packages:
+            - cmake
+            - cmake-data
+            - clang-6.0
+            - libboost-all-dev
+            - build-essential
+      env:
+        - MATRIX_EVAL="CC=/usr/local/clang-5.0.0/bin/clang && CXX=/usr/local/clang-5.0.0/bin/clang++"
+        - BUILD_TYPE="Release"
+        - BUILD_FLAGS=""
+        - CMAKE_FLAGS=""
+        - TEST_TASK="test"
+    - name: "Linux gcc release"
+      os: linux
+      compiler:
+        - gcc
+      sudo: required
+      addons:
+        apt:
+          update: true
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-7
+            - cmake
+            - cmake-data
+            - libboost-all-dev
+            - build-essential
+      env:
+        - MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
+        - BUILD_TYPE="Release"
+        - BUILD_FLAGS=""
+        - CMAKE_FLAGS='-DCMAKE_CXX_FLAGS_RELEASE="-O2"'
+        - TEST_TASK="test"
+    - name: clang-format
+      install:
+        - apt-cache search clang-format
+      addons:
+        apt:
+          sources: 
+            - 'llvm-toolchain-trusty'
+            - 'ubuntu-toolchain-r-test'
+          packages: 
+            - 'clang-format-8'
+      env:
+        - CLANG_FORMAT_BIN=clang-format-8
+        - TEST_TASK="format"
+    - name: "Build docker image"
+      services: 
+        - docker
+      env:
+        - TEST_TASK="docker"
+
+before_install:
+  - eval "${MATRIX_EVAL}"
+  - sudo add-apt-repository -y ppa:ubuntugis/ubuntugis-unstable
+  - sudo apt update 
+  - sudo apt install libgdal-dev gdal-bin
+
+script:
+  - if [ $TEST_TASK == 'test' ];
```

Comment:
I'd prefer to see it in a Makefile

Reply:
I agree. We should put all the different targets and test execution into the cmake file. And this reminds me, we should also add an installation target, so that 'make install' will work.

This will not replace the switch case statement though, since this is the only way to have completely different build methods in Travis matrix (or I was not able to find a better solution).

I will create a new issue for the additional cmake targets.

Reply:
Issue: https://github.com/heremaps/tin-terrain/issues/41